### PR TITLE
<LibNMath> Added new function to neomath library

### PR DESCRIPTION
**Pull request by akashkollipara**

### CODE REVIEW

**Comment by akashkollipara** [comment]
Do not close or proceed with merging this yet...
Planning to add Matrix manipulation algorithms.

Do should i raise another PR? @VisorFolks/cc-admin

**Comment by pranjalchanda08** [comment]
I can not see Jenkins details:
image

**Comment by akashkollipara** [comment]
yes you cannot see as u need the ngrok link...

**Comment by pranjalchanda08** [comment]
> yes you cannot see as u need the ngrok link... you can only see the status mark... details we cant.

Do is it because of continuous changing of links?

**Comment by akashkollipara** [comment]
> > yes you cannot see as u need the ngrok link... you can only see the status mark... details we cant.
> 
> So is it because of continuous changing of links?

Yes precisely!

File: jenkins.pipe [hunk 1]
```diff
@@ -15,6 +15,7 @@ pipeline {
 		stage('clean') {
 			steps {
 				sh 'make demo_avr clean'
+				sh 'rm -rf toolchain'
```
Comment (pranjalchanda08) [inline jenkins.pipe:18]: why do we need to remove the toolchain in clean?

File: src/lib/libnmath/nmath.c [hunk 2]
```diff
@@ -10,23 +10,75 @@
...
+ * @param[in] b: Input numer
+ *
+ * @return gcd: returns GCD of a, b
+ */
 unsigned long gcd(unsigned long a, unsigned long b)
 {
-	if(b == 0)
-		return a;
-	return gcd(b, (a % b));
+	return (b == 0) ? a: gcd(b, (a % b));
```
Comment (pranjalchanda08) [inline src/lib/libnmath/nmath.c:69]: Is it possible to not use recursion? for large value of a and b stack smash can occur.

**Review by pranjalchanda08 — CHANGES_REQUESTED** [review]

**Comment by pranjalchanda08** [comment]
> > > yes you cannot see as u need the ngrok link... you can only see the status mark... details we cant.
> > 
> > 
> > So is it because of continuous changing of links?
> 
> Yes precisely!

Might be post the logs of CI?

File: src/lib/libnmath/nmath.c (hunk 2 above)
Comment (akashkollipara) [reply src/lib/libnmath/nmath.c:69]: Well... this according to code consumes stack.
but with -Os flag we are trying to optimize use of memory.
So when you look at disassembly, this is just a while loop.

File: jenkins.pipe (hunk 1 above)
Comment (akashkollipara) [reply jenkins.pipe:18]: Why to retain when we can freshly checkout the TC for every build. It saves space on build disk also.

File: src/lib/libnmath/nmath.c (hunk 2 above)
Comment (pranjalchanda08) [reply src/lib/libnmath/nmath.c:69]: have u tested it with larger numbers?

File: jenkins.pipe (hunk 1 above)
Comment (pranjalchanda08) [reply jenkins.pipe:18]: Wont it take time to pull again and again?

File: src/lib/libnmath/nmath.c (hunk 2 above)
Comment (akashkollipara) [reply src/lib/libnmath/nmath.c:69]: image
This is disasm of function with long, long numbers.

File: jenkins.pipe (hunk 1 above)
Comment (akashkollipara) [reply jenkins.pipe:18]: nope... 7-8 sec is all it needs

File: src/lib/libnmath/nmath.c (hunk 2 above)
Comment (pranjalchanda08) [reply src/lib/libnmath/nmath.c:69]: ok then this makes sense.

**Review by pranjalchanda08 — APPROVED** [review]

**Pull request merged**